### PR TITLE
Add cmake buildsystem

### PR DESCRIPTION
**Pull request by notspiff**
This adds a cmake buildsystem. It is a somewhat stripped down version of my kodi-cmake stuff; in particular no depends / bootstrap (addons) support.

It is currently only working on linux, but i welcome all effort to make it work on other platforms. It should be well prepared for it.

Some advantages over the autotools system;

1) all builds are out-of-tree
2) tests are easily accessible (just execute 'make check')
3) make check-commits does so for a patch-series (see project/cmake/scripts/common/checkcommit.cmake for parameters).
(4) same buildsystem for all platforms - this needs help/effort from others to be realized)
(4) doxy accessible using 'make doc' - i had to disable this because it requires changes that affect the autotools system and i want to do them separately).

some of the things here are a bit rough, for instance ffmpeg/cguid handling was improvised as i had to replace it for the strip-down version.

use it by pointing cmake to kodi_root/project/cmake while in any folder != kodi_root.
build using 'make', then run ./kodi.bin straight from the build folder. all files needed to do so are mirrored in the build tree. some addons might be missing in this stripped down version.

### CODE REVIEW

**Comment by FernetMenta** [comment]
nice! I will try as soon I am back home and have a Linux system. What needs to be done to get this work for OSX? (my laptop is a mac)

**Comment by notspiff** [comment]
1) setup the arch file (see project/cmake/scripts/linux/archsetup.cmake - duplicate in e.g. project/cmake/scripts/osx/archsetup.cmake - depends on whatever you call the arch at project/cmake/script/common).
2) add osx specific files to build. see project/cmake/treedata/<common|linux|osx>. all text files are processed.
3) stuff i probably haven't though of..

**Comment by FernetMenta** [comment]
this leaves the old build system operational, right?

File: project/cmake/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+add_executable(JsonSchemaBuilder ${CORE_SOURCE_DIR}/tools/depends/native/JsonSchemaBuilder/src/JsonSchemaBuilder.cpp)
+add_subdirectory(${CORE_SOURCE_DIR}/tools/depends/native/TexturePacker build/texturepacker)
+
+# Codegen
+add_subdirectory(${CORE_SOURCE_DIR}/xbmc/interfaces/swig build/swig)
+
+# main binary
+add_executable(${APP_NAME_LC}.bin ${CORE_SOURCE_DIR}/xbmc/main/main.cpp $<TARGET_OBJECTS:compileinfo>)
+target_link_libraries(${APP_NAME_LC}.bin ${SYSTEM_LDFLAGS} -Wl,--whole-archive ${core_DEPENDS} -Wl,--no-whole-archive
+                      ${DEPLIBS} ${CMAKE_DL_LIBS})
```
Comment (wsnipex) [inline project/cmake/CMakeLists.txt]: line break

**Comment by wsnipex** [comment]
yes, this doesn't break autotools.

Been a while since I worked on this, but I still remember lots of it.
bout time to get this flying :+1:

**Comment by FernetMenta** [comment]
ok, let's get started.

**Pull request merged**